site_root/input/resources/extensions/StructureDefinition-us-core-ethnicity.json

```diff
@@ -0,0 +1,2133 @@
+  },
+  "url": "http://hl7.org/fhir/us/core/StructureDefinition/us-core-ethnicity",
+  "version": "3.1.0",
+  "name": "USCoreEthnicityExtension",
```

Comment:
Idk if we discussed a standard for `name`, but I've been populating it with the `id`. It's just supposed to be machine readable so I figured `id` will always be machine readable.

Reply:
So, for this one and similar ones below, they are as-is imports as you can find here: [StructureDefinition-us-core-ethnicity](https://www.hl7.org/fhir/us/core/StructureDefinition-us-core-ethnicity.json.html). For the participant profiles and the biospecimen conformance resources, I also used the IDs as their names. It's up to us whether we want to tweak these mere imports.

Reply:
I think its ok to tweak it because we've already copied in the actual US Core source rather than extend it.

Reply:
Hmmmm, on one hand since we've already copied the source and overwritten the `url` as a kids first one, I'm thinking maybe we should format the resource according to our standards. But on the other hand I'm wondering whether we should just leave it to preserve as much of the original source? Idk I'm torn...

@ShahimEssaid @fiendish any thoughts?

Reply:
> Hmmmm, on one hand since we've already copied the source and overwritten the url as a kids first one

??? Right there I see `  "url": "http://hl7.org/fhir/us/core/StructureDefinition/us-core-ethnicity"`. That's not overwritten.

Reply:
Ohh I thought we decided to use our own `url` because we're treating these as our own conf resources (since we're copying and pasting the source rather than extending the source).

Reply:
We decided to use their URLs as they are since it is a mere import.

Reply:
I think the URL should stay the same for interoperability when that becomes important. 

The short of the what's said below is that the "name" and probably "sliceName" should be what that extension would have been named if it was a property on the parent Resource, camelCased. 

Regarding naming, I can't think of a technical reason for doing it one way or the other. The FHIR spec uses the name as a property name (camel case) and the publisher uses that create a simplified Extension documentation page where it's simply showed as a property name. Extensions are basically complicated properties and their "id" and "name" conventions try to communicate this. See this example for the display of partOf: https://hl7.org/FHIR/extension-event-partof.html

In the above example, there is also a "title" element for the extension but the IG publisher seems to use the name rather than the title. Also, notice that the "id" is named with a resource type prefix and because this extension is specific to Patient it's prefixed with "patient-". 

Other extensions that are not resource specific have different conventions for the id. For example this one has the "workflow pattern" "event" as the prefix: https://hl7.org/FHIR/extension-event-partof.xml.html

On the other hand, this US Core simple extension doesn't seem to use the "name" value to create a simplified view in the IG: https://www.hl7.org/fhir/us/core/StructureDefinition-us-core-direct.html  Maybe this is all dependent on the template being used and has nothing specific with the publisher.

I briefly tried to find a complex extension in the FHIR spec to see how it's displayed but couldn't fine one. US Core has one and it appears that the "sliceName" is used in the display. The nested and inline extensions don't have name/title elements (but could be added) but it appears that the slice name is used: https://www.hl7.org/fhir/us/core/StructureDefinition-us-core-race.html